### ISSUE TITLE
Abort() must be triggered when an error condition (e.g., an Exception) is locally triggered on a process

### ISSUE THREAD

**Issue by amartinhuertas** [gridap/GridapDistributed.jl#17]
In order to avoid MPI deadlocks we must automatically call [`MPI.Abort()`](https://juliaparallel.github.io/MPI.jl/v0.12/environment.html#MPI.Abort) when an error condition (e.g., a failing test or an exception) is locally triggered on a processor. Problem: how can this be fulfilled when  an error condition is produced within an external dependency, such as, e.g., `Gridap.jl`? Perhaps using Julia built-in support for [exceptions](https://docs.julialang.org/en/v1/manual/control-flow/#Exception-Handling-1)? (i.e., try catch statements).

**Comment by fverdugo** [comment]
Yes, we need some context handling for the MPI library and I think exceptions are the way to go (even though they can be hidden from the user).

My question is if MPI.jl provides some kind of context handling ready to be used? something like
```julia

MPI.init() do

# My MPI code here

end

that guarantees that MPI.Finalize or MPI.Abourt will be called eventually, even if errors in the do-block are found.
``` 

Anyway, we can implement something like:

```julia
MPIPETScCommunicator() do comm

# GridapDistributed user code here

end
```

which internally does exception handling

**Comment by amartinhuertas** [comment]
solved in 53c0fc7969e2de7b17c45593614ed909642f45e5
